src/Symfony/Component/Yaml/Inline.php

```diff
@@ -660,7 +665,7 @@ private static function evaluateScalar($scalar, $flags, $references = [])
                     case 0 === strpos($scalar, '! '):
                         @trigger_error(self::getDeprecationMessage('Using the non-specific tag "!" is deprecated since Symfony 3.4 as its behavior will change in 4.0. It will force non-evaluating your values in 4.0. Use plain integers or !!float instead.'), E_USER_DEPRECATED);
 
-                        return (int) self::parseScalar(substr($scalar, 2), $flags);
+                        return (int) self::parseScalar((string) substr($scalar, 2), $flags);
```

Comment:
In PHP 7.4 substr('! ', 2) returns false (instead of '' previously). parseScalar expects a string.